src/web3py/extensions/lido_validators.py

```diff
@@ -175,23 +173,8 @@ def get_lido_validators_by_node_operators(self, blockstamp: BlockStamp) -> Valid
     def get_lido_node_operators(self, blockstamp: BlockStamp) -> list[NodeOperator]:
         result = []
 
-        for module in self.get_staking_modules(blockstamp):
-            operators = self.w3.lido_contracts.staking_router.functions.getAllNodeOperatorDigests(
-                module.id
-            ).call(block_identifier=blockstamp.block_hash)
-
-            for operator in operators:
-                result.append(NodeOperator.from_response(operator, module))
+        for module in self.w3.lido_contracts.staking_router.get_staking_modules(blockstamp.block_hash):
+            operators = self.w3.lido_contracts.staking_router.get_all_node_operator_digests(module, blockstamp.block_hash)
+            result.extend(operators)
 
         return result
-
-    @lru_cache(maxsize=1)
-    @list_of_dataclasses(StakingModule)
-    def get_staking_modules(self, blockstamp: BlockStamp) -> list[StakingModule]:
```

Comment:
I wouldn't remove this method, I used it in CSM.

Reply:
You can use new version directly from staking router contract
`w3.lido.staking_router.get_staking_modules`